File: src/test/java/toby/command/commands/music/ResumeCommandTest.java
package toby.command.commands.music;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import toby.command.CommandContext;

import static org.mockito.Mockito.*;

class ResumeCommandTest implements MusicCommandTest {

    private ResumeCommand resumeCommand;

    @BeforeEach
    void setup() {
        setupCommonMusicMocks();
        resumeCommand = new ResumeCommand();
    }

    @AfterEach
    void tearDown() {
        tearDownCommonMusicMocks();
    }

    @Test
    public void test_resumeMethod_withCorrectChannels_andPausableTrack() {
        //Arrange
        setUpAudioChannelsWithBotAndMemberInSameChannel();
        CommandContext commandContext = new CommandContext(event);
        when(interactionHook.sendMessage("Resuming: ")).thenReturn(webhookMessageCreateAction);
        when(audioPlayer.isPaused()).thenReturn(true);
        when(playerManager.isCurrentlyStoppable()).thenReturn(true);

        //Act
        resumeCommand.handleMusicCommand(commandContext, playerManager, requestingUserDto, 0);

        //Assert
        verify(interactionHook, times(1)).deleteOriginal();
        verify(event.getHook(), times(1)).sendMessage("Resuming: `");
        verify(webhookMessageCreateAction, times(1)).addContent("Title");
        verify(webhookMessageCreateAction, times(1)).addContent("` by `");
        verify(webhookMessageCreateAction, times(1)).addContent("Author");
        verify(webhookMessageCreateAction, times(1)).addContent("`");
        verify(audioPlayer, times(1)).isPaused();
        verify(audioPlayer, times(1)).setPaused(false);
    }
}

File: src/test/kotlin/toby/command/commands/music/ResumeCommandTest.kt
package toby.command.commands.music

import io.mockk.*
import net.dv8tion.jda.api.entities.MessageEmbed
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import toby.command.CommandContext
import toby.command.CommandTest
import toby.command.CommandTest.Companion.event
import java.awt.Color

internal class ResumeCommandTest : MusicCommandTest {
    private lateinit var resumeCommand: ResumeCommand

    @BeforeEach
    fun setup() {
        setupCommonMusicMocks()
        resumeCommand = ResumeCommand()
    }

    @AfterEach
    fun tearDown() {
        tearDownCommonMusicMocks()
        clearAllMocks()
    }

    @Test
    fun test_resumeMethod_withCorrectChannels_andPausableTrack() {
        // Arrange
        setUpAudioChannelsWithBotAndMemberInSameChannel()
        val commandContext = CommandContext(event)
        val audioPlayer = MusicCommandTest.mockAudioPlayer
        val playerManager = MusicCommandTest.playerManager
        val hook = event.hook
        every { audioPlayer.isPaused } returns true
        every { playerManager.isCurrentlyStoppable } returns true
        every { event.getOption("resume") } returns mockk {
            every { asInt } returns 1
        }



        // Act
        resumeCommand.handleMusicCommand(
            commandContext,
            playerManager,
            CommandTest.requestingUserDto,
            0
        )

        val embedSlot = slot<MessageEmbed>()
        // Assert
        verify(exactly = 1) { hook.sendMessageEmbeds(capture(embedSlot)) }
        verify(exactly = 1) { audioPlayer.isPaused = false }

        // Assert on the captured EmbedBuilder
        val messageEmbed = embedSlot.captured
        assert(messageEmbed.title == "Track Pause/Resume")
        assert(messageEmbed.description == "Resuming: `Title` by `Author`")
        assert(messageEmbed.color == Color.CYAN)
    }
}
